### ISSUE TITLE
azurerm_role_definition creating NEW role get error:RoleDefinitionDoesNotExist

### ISSUE THREAD

**Issue by brianmoritz** [terraform-providers/terraform-provider-azurerm#10479]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
When creating a new Azure role definition terraform apply will error: 
azurerm_role_definition.example: apply errored, but we're indicating that via the Error pointer rather than returning it: 
```
authorization.RoleDefinitionsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="RoleDefinitionDoesNotExist" Message="The specified role definition with ID '5............................' does not exist."
```
The role eventually shows up in the portal however it seems (my guess) like TF is checking for the role before the Azure API finishes creating it.

### Version
Tried multiple versions:
+ Terraform v0.14.4 & v0.14.5
+ provider registry.terraform.io/hashicorp/azuread v1.3.0
+ provider registry.terraform.io/hashicorp/azurerm v2.45.1 & v2.46.1

### Affected Resource(s)

azurerm_role_definition
data.azurerm_subscription

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

data "azurerm_subscription" "primary" {
}

resource "azurerm_role_definition" "example" {
  name        = "my-custom-role-bug-test"
  scope       = data.azurerm_subscription.primary.id
  description = "This is a custom role created via Terraform"

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [
    data.azurerm_subscription.primary.id,
  ]
}
```

### Debug Output
https://gist.github.com/brianmoritz/4174112ab02a7f0f259e7134e0fa0d18#file-terraform_log_scrub-log

### Expected Behaviour

New role is created and saved in state file.

### Actual Behaviour

New role is created after some time however terraform errors:
"The specified role definition with ID '5.........' does not exist."
Because of the error the role is not written to state.
Subsequent Terraform apply commands error that the role alreay exists. 

### Steps to Reproduce

t1. `terraform apply`

**Comment by ManikAggarwal** [comment]
In our case, role doesn't even show in portal. It just fails first time saying "RoleDefinition does not exist". Then in the subsequent run, it says "RoleDefinition exists... cannot update". However, role never shows up in the portal.

**Comment by gettek** [comment]
+1

**Comment by gettek** [comment]
Strangely enough, it works with targetting (most of the time) but this by no means is a scalable solution.
`-target="azurerm_role_definition.example"`

Introducing a random uuid resource to generate the role_definition_id appears to be a little more stable:

```hcl
resource random_uuid example {}

resource "azurerm_role_definition" "example" {
  name               = "my-custom-role-bug-test"
  role_definition_id = random_uuid.example.result
  scope              = data.azurerm_subscription.primary.id
  description        = "This is a custom role created via Terraform"

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [
    data.azurerm_subscription.primary.id,
  ]
}
```

Better still there may be a fix available soon as mentioned by @tombuildsstuff in #10434 🤞🏽

**Comment by gettek** [comment]
Unfortunately still getting this in version 2.46.1

**Comment by tombuildsstuff** [comment]
👋 

Thanks for opening this issue.

Taking a look through here this appears to be the same root-cause as #440, where the Azure API creates a second role definition during updates and then reconciles it a short while later - meaning when Terraform goes to look up the updated role definition (with an updated ID), it's gone (since it's been reconciled into the original). There's a fix for this in the form of #9850 - but we've not had time to come back to that just yet - but since the root cause for this appears to be the same as #440 (albeit during creation, not update) - I'm going to close this in favour of #440 for the moment - would you mind subscribing to that issue for updates?

Thanks!
